Sources/Build/BuildPlan/BuildPlan.swift

```diff
@@ -692,6 +727,203 @@ public class BuildPlan: SPMBuildCore.BuildPlan {
     }
 }
 
+extension BuildPlan {
+    /// Applies plugins to the given module as needed. Each plugin is passed an input context that provides
+    /// information about the module to which it is being applied (along with some information about that
+    /// module's dependency closure). The plugin is expected to generate an output in the form of commands
+    /// that will later be run before or during the build, and can also emit debug output and diagnostics.
+    ///
+    /// Each result returned by this function includes an ordered list of commands to run before the build
+    /// of the module, and another list of the commands to incorporate into the build graph so they run
+    /// at the appropriate times during the build.
+    ///
+    /// Any warnings and errors related to running the plugin will be emitted to `diagnostics` when
+    /// `surfaceDiagnostics` parameter is set to `true`.
+    ///
+    /// Note that warnings emitted by the the plugin itself will be returned in the PluginEvaluationResult
```

Comment:
Looks like the comment is outdated, there's no such type as `PluginEvaluationResult` anywhere in the codebase. It's probably referring to `BuildToolPluginInvocationResult`?

```suggestion
    /// Note that warnings emitted by the the plugin itself will be returned in the ``BuildToolPluginInvocationResult``
```

Reply:
✅